### ISSUE TITLE
[BUG] Connected to multiple peers but not syncing

### ISSUE THREAD

**Issue by swilkinson1** [Chia-Network/chia-blockchain#4971]
**Describe the bug**
My farmer (farmer, full node, harvester, wallet, daemon) running on Ubuntu 20.04.2 (chia version 1.1.6.dev0) stopped syncing overnight/this morning, and now won't sync even though I'm connected to multiple peers.  Has been stuck on block 278867 since syncing/farming stopped.

**To Reproduce**
Started my farmer with `chia start farmer`. Have tried a `chia stop all -d` and starting up again. Port 8444 is open.

**Expected behavior**
Should sync up when connected.

**Screenshots**
Output from `chia show -sc`
Current Blockchain Status: Full Node syncing to block 279875 
Currently synced to block: 278687
Current Blockchain Status: Not Synced. Peak height: 278687
      Time: Fri May 14 2021 08:11:15 UTC                  Height:     278687

Estimated network space: 4.423 EiB
Current difficulty: 304
Current VDF sub_slot_iters: 112721920
Total iterations since the start of the blockchain: 899949780394

  Height: |   Hash:
   278687 | 891792cff2c48d1b648817a28d7e7c147e6bf5ee541e3355c9ac08a5f7b4e0b4
   278686 | 496ab3dc5131d339f3e6237f4cb792864167f64e5ab266ba7062ba770c60a3d7
   278685 | 1ec7d0815db6ef180f1a3d1a0185bfe2b4d3b5d82d30990db0bd3e6b695ef067
   278684 | 1d2cedb5f357b47b2c3a5253917911e0a476c83e161b013f979fd0b7878dcda8
   278683 | 3d2168ffb56b65595b1d88d99f5006cc06b3d31fd1d3e9d1860c08326444d291
   278682 | 7782ad9204352353d4b98a11a0ffce127e4d936953203b5ea5ff9350d9d18b56
   278681 | 8f1874501055338cce05b0aa35cfa511297a787fbbe32f66234ed8a5b876be97
   278680 | 47b2c3ebb66432fba4e1064e347cbc089b530ce744e17d1206224999b76fe25c
   278679 | 65d27ae4a163e9f05a7e93fd25d7ebbbf6c30f21e637cd58a5a31a1505afe7b1
   278678 | 84f603e8fa662451ead412e034927ee880dd5274cc79fb9005bd5e4bd2661227

Connections:
Type      IP                                     Ports       NodeID      Last Connect      MiB Up|Dwn
WALLET    127.0.0.1                              34426/8449  ed2a1260... May 14 11:43:49      0.0|0.0    
FARMER    127.0.0.1                              34434/8447  38946635... May 14 11:43:51      0.0|0.0    
FULL_NODE 84.193.35.3                            37342/8444  1ef55227... May 14 13:28:49      0.0|0.0    
                                                 -SB Height:      297    -Hash: 02435862...
FULL_NODE 73.87.69.179                           52375/8444  5b5fc544... May 14 13:28:49     38.9|0.0    
                                                 -SB Height:     7371    -Hash: 9d59896c...
FULL_NODE 66.25.26.18                            65047/8444  bb1b66ca... May 14 13:30:16     44.7|0.0    
                                                 -SB Height:   259959    -Hash: 43b77e86...
FULL_NODE 186.250.161.61                         57068/8444  70698828... May 14 13:28:49      0.0|0.0    
                                                 -SB Height:   264970    -Hash: 2efc6ba0...
FULL_NODE 88.80.148.241                          49890/8444  3ce8e907... May 14 13:28:49      0.0|0.0    
                                                 -SB Height:        0    -Hash:  Info...
INTRODUCER introducer-va.chia.net                  8444/8444  0b1ad21d... May 14 12:46:22      0.0|0.0    
FULL_NODE 118.200.54.16                           8444/8444  c6efb1b1... May 14 13:28:49      0.0|0.0    
                                                 -SB Height:   200160    -Hash: 084453bf...
FULL_NODE 5.216.181.83                           49868/8444  7a434aff... May 14 13:28:49      0.0|0.0    
                                                 -SB Height:      160    -Hash: 853946df...
FULL_NODE 90.212.178.84                          61827/8444  1f5270a6... May 14 13:28:49      0.0|0.0    
                                                 -SB Height:    22336    -Hash: 7883bd3f...
FULL_NODE 114.35.64.203                           8444/8444  bb0d350d... May 14 13:30:34      0.0|14.8   
                                                 -SB Height:   279879    -Hash: 370b793c...
FULL_NODE 39.109.176.250                         50785/8444  840cb5c6... May 14 13:29:29      0.0|0.0    
                                                 -SB Height:        0    -Hash:  Info...

Output from `chia farm summary`:
Farming status: Not synced or not connected to peers

Some of these errors in my log:
2021-05-14T11:29:57.219 full_node full_node_server : INFO Connection closed: 66.75.0.206, node id: b23c644d661a25686978ae004d392f9e03312b9ca340ff99f34ea8fd2883047c
2021-05-14T11:29:57.219 full_node full_node_server : WARNING Banning 66.75.0.206 for 600 seconds
2021-05-14T11:29:57.220 full_node chia.full_node.full_node: INFO peer disconnected {'host': '66.75.0.206', 'port': 8444}
2021-05-14T11:29:57.221 full_node chia.full_node.full_node: ERROR Error with syncing: <class 'RuntimeError'>Traceback (most recent call last):
File "/home/ubuntu/chia-blockchain/chia/full_node/full_node.py", line 628, in _sync
raise RuntimeError(f"Weight proof did not arrive in time from peer: {weight_proof_peer.peer_host}")
RuntimeError: Weight proof did not arrive in time from peer: 66.75.0.206

But mostly my log is looking like this: 
2021-05-14T13:12:36.045 full_node full_node_server        : INFO     -> handshake to peer 135.181.145.43 c1ce09d9bdadbcb715d841b4a60897fc088ef8182fd65127e254da4355226793
2021-05-14T13:12:36.158 full_node full_node_server        : INFO     Connected with full_node {'host': '135.181.145.43', 'port': 8444}
2021-05-14T13:12:36.160 full_node full_node_server        : INFO     -> new_peak to peer 135.181.145.43 c1ce09d9bdadbcb715d841b4a60897fc088ef8182fd65127e254da4355226793
2021-05-14T13:12:36.162 full_node full_node_server        : INFO     Connection closed: 135.181.145.43, node id: c1ce09d9bdadbcb715d841b4a60897fc088ef8182fd65127e254da4355226793
2021-05-14T13:12:36.164 full_node chia.full_node.full_node: INFO     peer disconnected {'host': '135.181.145.43', 'port': 8444}
2021-05-14T13:12:37.550 full_node full_node_server        : INFO     <- request_blocks from peer dccf94f0c23139ab2bde2810133805eae7921c398fd68f275fc86e6a23cf9b7b 149.34.5.4
2021-05-14T13:12:37.682 full_node full_node_server        : INFO     -> respond_blocks to peer 149.34.5.4 dccf94f0c23139ab2bde2810133805eae7921c398fd68f275fc86e6a23cf9b7b
2021-05-14T13:12:44.154 full_node full_node_server        : INFO     <- request_blocks from peer bb1b66ca6b33bdb214c821fe61e87209490720a8987ac23b57f0c2277abab607 66.25.26.18
2021-05-14T13:12:44.285 full_node full_node_server        : INFO     -> respond_blocks to peer 66.25.26.18 bb1b66ca6b33bdb214c821fe61e87209490720a8987ac23b57f0c2277abab607
2021-05-14T13:12:45.848 full_node full_node_server        : INFO     <- request_blocks from peer dccf94f0c23139ab2bde2810133805eae7921c398fd68f275fc86e6a23cf9b7b 149.34.5.4
2021-05-14T13:12:45.977 full_node full_node_server        : INFO     -> respond_blocks to peer 149.34.5.4 dccf94f0c23139ab2bde2810133805eae7921c398fd68f275fc86e6a23cf9b7b
2021-05-14T13:12:47.607 full_node full_node_server        : INFO     -> handshake to peer 170.238.228.63 37529e68df57eb6911892b5334b846f93f3f02b4042f961d2676a33e7dc0e5f2
2021-05-14T13:12:47.751 full_node full_node_server        : INFO     Connected with full_node {'host': '170.238.228.63', 'port': 8444}
2021-05-14T13:12:47.752 full_node full_node_server        : INFO     -> new_peak to peer 170.238.228.63 37529e68df57eb6911892b5334b846f93f3f02b4042f961d2676a33e7dc0e5f2
2021-05-14T13:12:47.755 full_node full_node_server        : INFO     Connection closed: 170.238.228.63, node id: 37529e68df57eb6911892b5334b846f93f3f02b4042f961d2676a33e7dc0e5f2
2021-05-14T13:12:47.756 full_node chia.full_node.full_node: INFO     peer disconnected {'host': '170.238.228.63', 'port': 8444}
2021-05-14T13:12:54.529 full_node full_node_server        : INFO     <- request_blocks from peer dccf94f0c23139ab2bde2810133805eae7921c398fd68f275fc86e6a23cf9b7b 149.34.5.4
2021-05-14T13:12:54.663 full_node full_node_server        : INFO     -> respond_blocks to peer 149.34.5.4 dccf94f0c23139ab2bde2810133805eae7921c398fd68f275fc86e6a23cf9b7b

**Desktop (please complete the following information):**

- OS: Linux
- OS Ubuntu 20.04.2
- CPU: Intel Core i5-9400

**Comment by stipus** [comment]
Can you tell me if your chia database (.chia/mainnet/db directory) is on a spinning drive or a ssd ?

**Comment by swilkinson1** [comment]
The DB is on a SSD. All plots are on HDDs (spinning drives).

**Comment by JoJ123** [comment]
I'm facing exactly the same issue, when I restart the docker container, I get one sync block and then it's stalled again. 
I tried also to go back to older version, but nothing is helping at all...

**Comment by IIPoliII** [comment]
https://github.com/Chia-Network/chia-blockchain/issues/4949

**Comment by burak-dereli** [comment]
> chia farm summary

Hi, how do you write commands to the console? I just couldn't do it. I have that same problem. The only difference from your system is that I amd a4-4000 processor. There are 32 connections right now. But it just doesn't synchronize. If I can get the logs I'll compare them with yours.

**Comment by swilkinson1** [comment]
I'm running everything through the command line, without a GUI. So I'm entering commands directly in a terminal window.  CLI reference is here: [https://github.com/Chia-Network/chia-blockchain/wiki/CLI-Commands-Reference](https://github.com/Chia-Network/chia-blockchain/wiki/CLI-Commands-Reference)

Logs are in ~/.chia/mainnet/logs/.  I also have mine turned up to INFO level logging.

**Comment by Maronee** [comment]
Hey, I am running into the same issues, for weird reasons the chain runs out of sync and previously the trick was to connect to a full node for an update, e.g. what workes was node.chia.net:8443

But ever since 1.1.4 this doesnt work anymore, when checking the logs, you can see that it connects and immedeatly disconnects, see this log:

```
2021-05-14T18:40:42.520 full_node full_node_server        : INFO     -> handshake to peer node.chia.net 6170cf3fada1c95024e56855db562f94ee52008749ea289b6cf9b7815da677b4
2021-05-14T18:40:42.556 full_node full_node_server        : INFO     Connected with full_node {'host': 'node.chia.net', 'port': 8444}
2021-05-14T18:40:42.557 full_node full_node_server        : INFO     -> new_peak to peer node.chia.net 6170cf3fada1c95024e56855db562f94ee52008749ea289b6cf9b7815da677b4
2021-05-14T18:40:42.558 full_node full_node_server        : INFO     Connection closed: node.chia.net, node id: 6170cf3fada1c95024e56855db562f94ee52008749ea289b6cf9b7815da677b4
```

That happens with every connection and then at some point in time it stays connected and syncs...

any ideas?

**Comment by tibbon** [comment]
I'm also experiencing this! I'm able to manually add peers to sync them, but then when they get close to done they disappear. It doesn't seem to be finding them automatically, and many I'm unable to connect to.

**Comment by Sokmunki** [comment]
I am also having the same issue.

I have tried a few things, 

- deleting blockchain database and peer database
- deleting just the block chain database and restoring from a point when i wasn't having issues.
- deleting the peer database
- rolling back to version 1.1.4
- adding official introducers as peers

One thing i've noticed, is that whilst syncing after restart or these deletions, is that i'm connected to multiple peers that are up to date, using chiastatus.com as my point of reference. But some time after reaching sync it starts to lag behind, i notice a few 'unfinished' header hash in the blocks 'pane' on the 'full node' window. Sometimes it recovers, but other times it ends up falling a few minutes behind.

I have port forward enabled, and had no issues previously. Was running on 1.1.4 when i started, about 10 days ago, and updated to 1.1.5 maybe 2 days ago.

**Comment by cyborgxxl** [comment]
> I am also having the same issue.
> 
> I have tried a few things,
> 
>     * deleting blockchain database and peer database
> 
>     * deleting just the block chain database and restoring from a point when i wasn't having issues.
> 
>     * deleting the peer database
> 
>     * rolling back to version 1.1.4
> 
>     * adding official introducers as peers
> 
> 
> One thing i've noticed, is that whilst syncing after restart or these deletions, is that i'm connected to multiple peers that are up to date, using chiastatus.com as my point of reference. But some time after reaching sync it starts to lag behind, i notice a few 'unfinished' header hash in the blocks 'pane' on the 'full node' window. Sometimes it recovers, but other times it ends up falling a few minutes behind.
> 
> I have port forward enabled, and had no issues previously. Was running on 1.1.4 when i started, about 10 days ago, and updated to 1.1.5 maybe 2 days ago.

Same problem here.... This suxx a lot.  Does the 1.1.4 work better than 1.1.5 or does it have the same issue?

**Comment by zeybor** [comment]
> * OS: Linux
> * OS Ubuntu 20.04.2
> * CPU: Intel Core i5-9400

I am having pretty much the same issues but on a windows machine. The only fix appears to be a restart. After the restart everything works okay for a few hours, until I get unsynced again; after which the only solution is another restart...

**Comment by Sokmunki** [comment]
> Same problem here.... This suxx a lot. Does the 1.1.4 work better than 1.1.5 or does it have the same issue?

I was hoping it would, but it has proven no better. seemed to work for a while and then same problem returns.

**Comment by HudsonProdigy** [comment]
try closing gui. right click and run as admin.

**Comment by Sokmunki** [comment]
Ok, thought I'd give an update. 

TL;DR only out-of-sync while farming, if no plot directory added, stays in perfect sync.

Last night after deciding that 1.1.4 was not much better than 1.1.5, I uninstalled 1.1.4, deleted the directories and re-installed 1.1.5. 

Being a Saturday night, I had a few drinks and forgot about it until this morning. When I checked, it was in perfect sync, like some kind of Christmas miracle. But then to my horror, I noticed that I forgot to add my plot directory. So all night it wasn't farming. 

I added my plots, made some breakfast and went out on a quick errand and by the time I came back, about 30 minutes later, it was out of sync again!

As an further experiment, I removed my plot directory and within the space of 10 minutes it was back in sync.

**Comment by fabioyy** [comment]
chia stop wallet
and then chia start node
seemns to get things done ( but i ḿ  not sure if wallet is required to farming )

**Comment by voycey** [comment]
I have tried everything, stopping the wallet and just keeping the node running, nothing syncs - just feels like Chia is VERY broken right now

**Comment by Sokmunki** [comment]
It appears, after limited testing anyway, that the issue in my case was the windows client. 

I ran up a new VM, installed Ubuntu 20.04.2 LTS, and it's stayed in sync for about 4 hours so far.

I followed the instructions for installation as detailed in the wiki. 

The only changes I've made are, set enable_upnp to false and Under full_node, I changed introducer_peer host to my closest introducer, which in my case is, introducer-apse.chia.net. 

I copied my mainnet db from my other machine, so that didn't need to sync. But I see now that the wallet also has a db, which I didn't copy. So that is currently not synced.

**Comment by zeybor** [comment]
As for me, it seems that the issue was plotting, probably using too much of system resources. When I got unsynced, I just stopped plotting, and relaunch the GUI, and I was synched again.

**Comment by elforesto** [comment]
I was running into this issue when my DB was on shared HDD storage. Moving it to SSD resolved the issue and greatly improved performance.

**Comment by fl1ck3r** [comment]
> I was running into this issue when my DB was on shared HDD storage. Moving it to SSD resolved the issue and greatly improved performance.

In my case, I had to move my DB from the micro sd card on my pi4, to an physical HDD surprisingly. This solved my issue. And strangely enough I had a node only running on Linode, with an SSD, that had the same issue of slow database. I ended up removing the node from the instance, and rather using it to open my local pi4 port via wire guard. As I was not too happy opening my public ip.
